common/endpoints/endpoints.go

```diff
@@ -60,9 +73,56 @@ func (s *TwitterServer) statsHandler(w http.ResponseWriter, r *http.Request) {
 	const contentTypeVal = "application/json; charset=utf-8"
 	w.Header().Set(contentTypeHdr, contentTypeVal)
 
+	// get stats from this Twitter Server server
+	serviceStats := s.Stats.Render(false)
+
+	var allStats map[string]float64
+	json.Unmarshal(serviceStats, &allStats)
+
+	// get stats for the sidecars
+	sidecarStatsCh := make(chan []byte, len(s.SidecarStatPorts))
+	var wg sync.WaitGroup
+	for _, p := range s.SidecarStatPorts {
+		wg.Add(1)
+		go func(port int) {
+			defer wg.Done()
+			// sidecars must be running on the same machine and also be twitter servers
+			rsp, err := http.Get(fmt.Sprintf("http://localhost:%v/admin/metrics.json", port))
+			if err == nil {
+				defer rsp.Body.Close()
+				body, err := ioutil.ReadAll(rsp.Body)
+
+				if err == nil {
+					sidecarStatsCh <- body
+
+				} else {
+					log.Printf("ERROR: Reading Body of Stats Response for Sidecar at Port %v, Error: %v", port, err)
+				}
+			} else {
+				log.Printf("ERROR: Getting Stats for Sidecar at Port %v, returned an error %v", port, err)
+			}
+		}(p)
+	}
+
+	// wait until we have all the stats then combine
+	wg.Wait()
+	close(sidecarStatsCh)
+	for statsRsp := range sidecarStatsCh {
+		json.Unmarshal(statsRsp, &allStats)
```

Comment:
Arrays don't keep existing elements but maps do? I'm surprised enough by this behavior that it may warrant a comment despite it being a std lib.